### DISCUSSION TITLE
Rank mismatch between Expression and FunctionSpace when using IntervalMesh

### DISCUSSION THREAD

**Discussion by JunmingDuan** [firedrakeproject/firedrake#2576] [Firedrake support] [answered]
Dear all,

I tried to do interpolation on a 1D mesh:

mesh = IntervalMesh(10, 1)
V = FunctionSpace(mesh, "CG", 1)
x = SpatialCoordinate(mesh)
f = Function(V)
f.interpolate(x*2)

it reported: "raise RuntimeError('Rank mismatch: Expression rank %d, FunctionSpace rank %d'
RuntimeError: Rank mismatch: Expression rank 1, FunctionSpace rank 0"

Do you have any idea about this?
I have implemented a 2D test without such a problem.
Thanks for any help!

**Comment by ksagiyam** [comment, marked as answer]
You need `x, = SpatialCoordinate(mesh)` (just as in 2D/3D).

**Reply by JunmingDuan** [reply]
Thanks a lot!
